integration-tests/test/shared/utils.ts

```diff
@@ -166,3 +173,44 @@ export const waitForL2Geth = async (
   }
   return injectL2Context(provider)
 }
+
+export const awaitCondition = async (
```

Comment:
I think this should be in `core-utils`. We have a similar function here that could be replaced or merged with this one: https://github.com/ethereum-optimism/optimism/blob/25014cbc7a74b3cfc52462d3e765c4b4e5c39359/packages/contracts/src/hardhat-deploy-ethers.ts#L8-L26

Reply:
Cool, I can extract this into core utils.